### ISSUE TITLE
LSTM Object Detection Logits and Lables Shape

### ISSUE THREAD

**Issue by Shruthi-Sampathkumar** [tensorflow/models#7288]
### System information
- **What is the top-level directory of the model you are using**: Lstm object detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Colab
- **TensorFlow installed from (source or binary)**: preinstallled in colab
- **TensorFlow version (use command below)**: presintalled in colab
- **Bazel version (if compiling from source)**: 
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**: %run lstm_object_detection/train.py

Description of the Problem:

I have 2 non-background classes. I have converted my input to tfrecord which has indices 1 and 2 representing the labels (it is not one hot encoded). num_classes is set to 2 in config file. add_background_class is set to True in box_predictor_builder.py in Object Detection/builders/. I am getting shape issues between logits (40,324,3) and labels (40,324,2) when I run train.py from Lstm Object Detection. 

When I disable add_background_class and change num_class to 3, with my input tfrecord containing index 0 in addition to 1 and 2, the loss is constantly increasing. Also when I tried with num_class = 2 with my tfrecord containing only indices 1 and 2, the loss is stuck at 0.409 with no increase or decrease. Inference : Setting add_background_class to False makes training run with increasing loss values. 

I am not sure if this is a bug.
The stack trace shows "logits and labels must have the same shape ((40, 324, 3) vs (40, 324, 2))" .

Thanks in advance.
